src/host/ledger.h

```diff
@@ -947,7 +1064,12 @@ namespace asynchost
 
       if (require_new_file)
       {
-        files.push_back(std::make_shared<LedgerFile>(ledger_dir, last_idx + 1));
+        size_t start_idx = last_idx + 1;
+        bool is_recovery = recovery_start_idx.has_value() &&
+          start_idx > recovery_start_idx.value();
```

Comment:
`.value_or(0)`?

Reply:
This is setting the `is_recovery` boolean to true if `recovery_start_idx` is set and the value of `start_idx` is greater than `recovery_start_idx.value()`. I think what you're proposing is setting `is_recovery` to the value of `start_idx`, which isn't the same thing. What am I missing?

Reply:
No, that's not what I meant - I'm just proposing the shorter `bool is_recovery = start_idx > recovery_start_idx.value_or(start_idx);`